harmony_netcdf_to_zarr/mosaic_utilities.py

```diff
@@ -0,0 +1,195 @@
+""" A module including functions to take many input NetCDF-4 datasets and
+    create aggregated dimensions, allowing for a single Zarr output.
+
+"""
+from datetime import timedelta
+from typing import List, Optional, Union
+
+from dateutil.parser import parse as parse_datetime
+
+from netCDF4 import Dataset, Group, Variable
+import numpy as np
+
+
+NetCDF4Attribute = Union[bytes, str, np.integer, np.floating, np.ndarray]
+
+seconds_delta = timedelta(seconds=1)
+minutes_delta = timedelta(minutes=1)
+hours_delta = timedelta(hours=1)
+days_delta = timedelta(days=1)
+
+time_delta_to_unit_map = {seconds_delta: 'seconds',
+                          minutes_delta: 'minutes',
+                          hours_delta: 'hours',
+                          days_delta: 'days'}
+
+time_unit_to_delta_map = {'seconds': seconds_delta,
+                          'second': seconds_delta,
+                          'secs': seconds_delta,
+                          'sec': seconds_delta,
+                          's': seconds_delta,
+                          'minutes': minutes_delta,
+                          'minute': minutes_delta,
+                          'mins': minutes_delta,
+                          'min': minutes_delta,
+                          'hours': hours_delta,
+                          'hour': hours_delta,
+                          'hrs': hours_delta,
+                          'hr': hours_delta,
+                          'h': hours_delta,
+                          'days': days_delta,
+                          'day': days_delta,
+                          'd': days_delta}
+
+
+class DimensionsMapping:
```

Comment:
This class will be extended in DAS-1375, in which the aggregation of the same dimensions from each input NetCDF-4 file into a single output dimension will be added. Initial thoughts on that are that `self.output_dimensions` will be a mapping from the dimension name to a single `DimensionInformation` instance, containing the output dimension values, epochs and time units. I think there will be a new internal class method that produces these, and is called in the `__init__` of the class, but after `_map_input_dimensions`.